portal-api/src/v1/gef/controllers/facilities.controller.js

```diff
@@ -37,7 +37,7 @@ exports.create = async (req, res) => {
 
     const facilityParameters = { ...req.body, auditRecord: generateAuditDatabaseRecordFromAuditDetails(auditDetails) };
 
-    const facilityToInsert = new Facility(facilityParameters, existingDeal.version);
+    const facilityToInsert = new Facility(facilityParameters, existingDeal?.version);
```

Comment:
Is this needed? 

We check `existingDeal` is truthy above

Reply:
In the cases below that `existingDeal` doesnt exist, I think I'd prefer to handle it by returning a 404

Reply:
Good spot, not needed in that case. 

For the other cases, I'm a bit torn - I felt that we'd want the response to be returned regardless of if there's an issue finding the existing deal / getting the version number. So I guess as it is below, if there was an issue with the existing deal, it would pass it through as `undefined` into `facilitiesValidation` and by default not make the facility end date fields required... but maybe that isn't desired behaviour either? What do you think?

Reply:
I've added in that 404 handling now, let me know what you think!